pgxn/neon/pagestore_smgr.c

```diff
@@ -617,7 +921,32 @@ neon_wallog_page(SMgrRelation reln, ForkNumber forknum, BlockNumber blocknum, ch
 void
 neon_init(void)
 {
-	/* noop */
+	HASHCTL info;
+
+	if (MyPState != NULL)
+		return;
+
+	MyPState = palloc0(sizeof(PrefetchState));
+	
+	MyPState->bufctx = SlabContextCreate(TopMemoryContext,
+										 "NeonSMGR/prefetch",
+										 SLAB_DEFAULT_BLOCK_SIZE * 17,
```

Comment:
Where does the "17" come from?

Reply:
Hmm, I think `PS_GETPAGERESPONSE_SIZE * 16` would be more understandable then.

I wonder if using the slab allocator is just a premature optimization. Compared to copying 8kB of data to the buffer and all the other processing we do on each response, the overhead of one malloc+free is not that big. Did you see malloc/free in profiles?

Or we could redefine `pageserver_receive` so that the caller passes a StringInfo, and the function writes the response to that, enlarging it if needed. Then we could keep using the same buffer, skipping the palloc+free overhead altogether. That's a slightly larger refactoring though. If we're going to do that, then it would probably make sense to try to avoid the memcpy we currently do in `nm_unpack_response`. So probably best to leave that for another PR...